src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -324,10 +325,16 @@ def update_dashboard_filters(self, instance, validated_data):
         page_filter_keys = ["environment", "period", "start", "end", "utc"]
         dashboard_filter_keys = ["release"]
 
+        filters = {}
+
         if "projects" in validated_data:
-            instance.projects.set(validated_data["projects"])
+            if validated_data["projects"] == ALL_ACCESS_PROJECTS:
+                filters["all_projects"] = True
+            else:
+                if instance.filters and instance.filters.get("all_projects"):
+                    instance.filters["all_projects"] = False
+                instance.projects.set(validated_data["projects"])
```

Comment:
Do we need to clear instance.projects if all projects is selected?

Reply:
Yes! Good catch. I've added an `instance.projects.clear()` when setting `all_projects` to `True`